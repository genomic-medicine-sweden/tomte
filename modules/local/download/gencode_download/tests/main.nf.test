nextflow_process {

    name "Test Process GENCODE_DOWNLOAD"
    script "../main.nf"
    process "GENCODE_DOWNLOAD"
    tag "modules"
    tag "modules_local"
    tag "gencode_download"

    test("Download fasta -stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = 'GRCh38'
                input[1] = 46
                input[2] = 'fasta'
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Download gtf -stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = 'GRCh38'
                input[1] = 46
                input[2] = 'gtf'
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
