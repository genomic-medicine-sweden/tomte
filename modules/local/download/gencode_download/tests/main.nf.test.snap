{
    "Download gtf -stub": {
        "content": [
            {
                "0": [
                    "GRCh38.primary_assembly.genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "gencode.v46.primary_assembly.annotation.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,a809582924251f125611d2c774f1436e"
                ],
                "fasta": [
                    "GRCh38.primary_assembly.genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "gtf": [
                    "gencode.v46.primary_assembly.annotation.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,a809582924251f125611d2c774f1436e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T08:03:56.832995"
    },
    "Download fasta -stub": {
        "content": [
            {
                "0": [
                    "GRCh38.primary_assembly.genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "gencode.v46.primary_assembly.annotation.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,a809582924251f125611d2c774f1436e"
                ],
                "fasta": [
                    "GRCh38.primary_assembly.genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "gtf": [
                    "gencode.v46.primary_assembly.annotation.gtf:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,a809582924251f125611d2c774f1436e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T08:03:49.440695"
    }
}